### DISCUSSION TITLE
Suggest follow-up questions

### DISCUSSION THREAD

**Discussion by eric-sf** [microsoft/PubSec-Info-Assistant#413] [Q&A] [answered]
I see an option for "Suggest follow-up questions" in "Adjust". When I enable it I don't see any changes, what is that option supposed to do? Is there a way to get follow-up questions similar to bing chat?
![Screenshot 2023-12-20 at 4 11 08 PM](https://github.com/microsoft/PubSec-Info-Assistant/assets/11526292/b862f667-cef2-4044-bf57-abd83e403dca)

**Comment by ArpitaisAn0maly** [comment]
Hello [eric-sf](https://github.com/eric-sf),

follow up questions works as it is seen in SS below. if the question asked doesn't have a likely follow up it won't suggest any. In other words if answer is definitive it won't suggest follow up questions.
![image](https://github.com/microsoft/PubSec-Info-Assistant/assets/55892521/6998d42f-2fe2-43a2-aeea-9efe621897bc)

**Reply by eric-sf** [reply]
@ArpitaisAn0maly thanks for the reply, how do we determine "if answer is definitive"? Can I make the system always return follow-up question?

**Comment by dayland** [comment, marked as answer]
With the checkmark enabled in the "Adjust" panel. Prompts are added to request the LLM to return any suggested follow up questions. You can find this in our prompt engineering code in https://github.com/microsoft/PubSec-Info-Assistant/blob/v1.0/app/backend/approaches/chatreadretrieveread.py. 

In order to make this permanent this you would need to consider specific prompt engineering to reenforce this behavior as part of the system prompt or update the `follow_up_questions_prompt_content` langauge in the above file.
